app/js/components/labOrderEntry/LabEntryForm.jsx

```diff
@@ -1,70 +1,81 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
 import LabPanelFieldSet from './LabPanelFieldSet';
 import LabTestFieldSet from './LabTestFieldSet';
-import LabDraftOrder from './DraftOrder/LabDraftOrder';
+import LabDraftOrder from './LabDraftOrder';
 import ActiveOrders from './ActiveOrders';
 import PastOrders from './PastOrders';
 import Accordion from '../accordion';
 import { labCategories } from './labData';
-import { addDraftLabOrders, deleteDraftLabOrder, toggleDraftLabOrdersUgency } from '../../actions/draftLabOrderAction';
+
+import {
+  addTestPanelToDraft,
+  removeTestPanelFromDraft,
+  removeTestFromDraft,
+  addTestToDraft,
+  toggleDraftLabOrdersUgency,
+} from '../../actions/draftLabOrderAction';
+
 import { activeOrders, pastOrders } from './ordersHistoryMockData';
 import '../../../css/grid.scss';
 
-export class LabEntryForm extends React.Component {
+export class LabEntryForm extends PureComponent {
   state = {
     categoryId: 1,
-    defaultTests: [],
-    selectedTests: [],
     selectedPanelIds: [],
+    selectedPanelTestIds: [],
   };
 
-  selectTest = (test) => {
-    const { addDraftLabOrdersToStore, deleteDraftLabOrderFromStore } = this.props;
-    const selectedIds = this.state.selectedTests.map(currentTest => currentTest.id);
-    const defaultTestIds = this.state.defaultTests.map(currentTest => currentTest.id);
-    if (selectedIds.includes(test.id)) {
-      this.setState({
-        selectedTests: this.state.selectedTests.filter(currentTest => currentTest.id !== test.id),
-      });
-      deleteDraftLabOrderFromStore([test]);
-    } else if (!defaultTestIds.includes(test.id)) {
-      this.setState({ selectedTests: [...this.state.selectedTests, test] });
-      addDraftLabOrdersToStore([test]);
-    }
-  };
+  componentWillReceiveProps(nextProps) {
+    const {
```

Comment:
using componentWillReceiveProps lifecycle is not advised as it causes bugs and inconsistencies, therefore, it will be deprecated in React 17, more information [here](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops).

use `componentDidUpdate` instead

Reply:
@Efosaok that's why I changed it to a `React.PureComponent` instead of just `React.Component`. bcos using PureComponent handles the `didUpdate` logic by default so this is safe to use here and we ain't using React 17.